src/Microsoft.Diagnostics.Monitoring.EventPipe/Triggers/AspNet/AspNetRequestStatusTrigger.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.AspNet
+{
+    internal sealed class AspNetRequestStatusTrigger : AspNetTrigger<AspNetRequestStatusTriggerSettings>
+    {
+        private SlidingWindow _window;
+        private HashSet<int> _statusCodes = new();
+        private List<(int Min, int Max)> _statusCodeRanges = new();
+
+        public AspNetRequestStatusTrigger(AspNetRequestStatusTriggerSettings settings) : base(settings)
+        {
+            _window = new SlidingWindow(settings.SlidingWindowDuration);
+
+            foreach(string codeOrRange in settings.StatusCodes)
```

Comment:
Ideally, the trigger wouldn't have to do this again since we've parsed and validated the structure of the data already. See my `StatusCodeRange[]` comment.